source/s3_meta_request.c

```diff
@@ -1520,3 +1523,22 @@ void aws_s3_meta_request_result_clean_up(
 
     AWS_ZERO_STRUCT(*result);
 }
+
+bool aws_s3_meta_request_check_response_checksum_algorithm(
```

Comment:
nit: naming is a bit awkward here. aws_s3_meta_request_checksum_config_has_algorithm?